src/ucp/tag/tag_recv.c

```diff
@@ -143,16 +129,27 @@ static UCS_F_ALWAYS_INLINE ucs_status_ptr_t ucp_tag_recv_common(
     req->recv.op_attr       = param->op_attr_mask;
     req->recv.tag.tag       = tag;
     req->recv.tag.tag_mask  = tag_mask;
-    if (param->op_attr_mask & UCP_OP_ATTR_FIELD_CALLBACK) {
-        req->recv.tag.cb = param->cb.recv;
```

Comment:
to confirm, `req->recv.tag.cb` is never checked for NULL/used unless `UCP_REQUEST_FLAG_CALLBACK` is set on `req->flags`?

Reply:
yes, if UCP_REQUEST_FLAG_CALLBACK  is not set the CB should not be accessed at all